railties/lib/rails/api/generator.rb

```diff
@@ -0,0 +1,29 @@
+require "sdoc"
+
+class RDoc::Generator::API < RDoc::Generator::SDoc # :nodoc:
+  RDoc::RDoc.add_generator self
+
+  PUBLIC_CONSTANTS = %w( AbstractController ActionCable ActionController
+                         ActionDispatch ActionMailer ActionView ActiveJob
+                         ActiveModel ActiveRecord ActiveSupport Mime
+                         LoggerSilence Rails SourceAnnotationExtractor )
```

Comment:
Yeah, you're right, it's far from ideal !

Actually, it was fairly easy to determine whether a class was a core one or not, thus we don't have to maintain any list ! :-)